docs/pages/x/api/data-grid/selectors.json

```diff
@@ -391,7 +391,7 @@
   },
   {
     "name": "gridVisibleColumnDefinitionsSelector",
-    "returnType": "GridStateColDef[]",
+    "returnType": "(GridListViewColDef & { computedWidth: number })[]",
```

Comment:
This doesn't look right.
If anything, it should be `GridStateColDef[] | GridListViewColDef[]`.

Reply:
Hmm yes, it's returning the intersection type that can represent both return values. We could manually set the return type of the selector to `GridStateColDef[] | GridListViewColDef & { computedWidth: number }[]`... not sure if there's a better option?

Reply:
I think we can't change the return type of this selector.
How about a hack, where in listView mode we return `GridListViewColDef` enhanced with a bunch of fields from `GridStateColDef` set to `undefined`? If that's possible, we can keep the return type as `GridStateColDef[]`.

Reply:
All the fields outside of the `GridListViewColDef` and `GridStateColDef` intersection already seem to be optional (the only required coldef field is `field: string`), so I think we could just do a cast, no need to add actual `undefined` values. The only problem is `computedWidth: number`. I would add a selector like `gridListViewStateColDef` to transform the listview column to a normal `GridStateColDef` (even using the proper `computedWidth`, which would be `dimensions.root.width` I think). Tbh that's how I would have implemented listView to start with, just turn it into a normal column and let the existing logic work as before.

Reply:
> I think we could just do a cast, no need to add actual undefined values

Right 👍🏻

> The only problem is computedWidth: number. I would add a selector like gridListViewStateColDef to transform the listview column to a normal GridStateColDef

There already is `gridListColumnSelector` which returns `GridListViewState`:
`GridListViewState = (GridListViewColDef & { computedWidth: number }) | undefined`
So casting this one totally makes sense to me 👍🏻